sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -351,6 +371,7 @@ impl Target for HyperdebugSpiTarget {
 
     fn run_transaction(&self, transaction: &mut [Transfer]) -> Result<()> {
         let mut idx: usize = 0;
+        self.do_assert_cs(true)?;
```

Comment:
I believe the same can be achieve with refactoring of the counting (and the three level assert_cs(), do_assert_cs(), _do_assert_cs()), by keeping the existing implementation of assert_cs(), and do_assert_cs(), and adding code to run_transaction to the effect of:
```
   if cs_asserted_count == 0 {
      self.do_assert_cs(true)?;
   }
   [... do actual work ...]
   if cs_asserted_count == 0 {
      self.do_assert_cs(false)?;
   }
```
With this, the `cs_asserted_count` is really "asserted count outside of run_transaction", as the variable is not actually incremented during the transaction, but it works the same.

Reply:
This is perhaps style, but isn't it better to keep  the counting and manipulation of the CS signal in the same place?  This change simplifies the handling of CS counting within the transaction function and the `assert_cs()` public API and the `drop` method of the returned token.